demo/demo/tests/test_paddle.py

```diff
@@ -316,11 +313,14 @@ def test__payment_flow__trial_period(trial_period, paddle, user, user_client, pl
     payment.status = SubscriptionPayment.Status.PENDING
     payment.save()
 
-    check_unfinished_payments(within=timedelta(hours=1))
-
-    payment = SubscriptionPayment.objects.latest()
+    # Retry a few times to give paddle the time to update the tranaction status
+    for attempt in Retrying(wait=wait_incrementing(start=2, increment=2), stop=stop_after_attempt(20)):
+        with attempt:
+            check_unfinished_payments(within=timedelta(hours=1))
+            payment = SubscriptionPayment.objects.latest()
+            if payment.status != SubscriptionPayment.Status.COMPLETED:
+                raise TryAgain()
```

Comment:
Since it's the same block of code used in `test__payment_flow__regular` and `test__payment_flow__trial_period`, you could extract it to separate function and put it into `helpers.py`

Reply:
Hi I removed one by mistake but these are two different ones:
```python

    # ---- test_payment_status_endpoint_post ----
    for attempt in Retrying(wait=wait_incrementing(start=2, increment=2), stop=stop_after_attempt(10)):
        with attempt:
            response = user_client.post(f'/api/payments/{payment.id}/')
            assert response.status_code == 200, response.content
            result = response.json()
            if result['status'] != 'completed':
                raise TryAgain()
```

and 
```python
    # Retry a few times to give paddle the time to update the tranaction status
    for attempt in Retrying(wait=wait_incrementing(start=2, increment=2), stop=stop_after_attempt(20)):
        with attempt:
            check_unfinished_payments(within=timedelta(hours=1))
            payment = SubscriptionPayment.objects.latest()
            if payment.status != SubscriptionPayment.Status.COMPLETED:
                raise TryAgain()
```

I'm not sure whether it would make sense to put the automate_payment function in the `helpers.py` module as it exposes logic that seems exclusively used by paddle.

Reply:
I agree, you are right